bin/fuel-core/src/cli/snapshot.rs

```diff
@@ -230,33 +236,39 @@ fn full_snapshot(
     let prev_chain_config = load_chain_config(prev_chain_config)?;
     writer.write_chain_config(&prev_chain_config)?;
 
-    fn write<T>(
-        db: &CombinedDatabase,
+    fn write<T, DbDesc>(
+        db: &Database<DbDesc>,
         group_size: usize,
         writer: &mut SnapshotWriter,
     ) -> anyhow::Result<()>
     where
-        T: TableWithBlueprint<Column = <OnChain as DatabaseDescription>::Column>,
-        T::Blueprint: BlueprintInspect<T, Database>,
+        T: TableWithBlueprint<Column = <DbDesc as DatabaseDescription>::Column>,
+        T::Blueprint: BlueprintInspect<T, Database<DbDesc>>,
         TableEntry<T>: serde::Serialize,
         StateConfigBuilder: AddTable<T>,
+        DbDesc: DatabaseDescription,
     {
-        db.on_chain()
-            .entries::<T>(None, IterDirection::Forward)
+        db.entries::<T>(None, IterDirection::Forward)
             .chunks(group_size)
             .into_iter()
             .try_for_each(|chunk| writer.write(chunk.try_collect()?))
     }
     let group_size = encoding.group_size().unwrap_or(MAX_GROUP_SIZE);
 
-    write::<Coins>(&db, group_size, &mut writer)?;
-    write::<Messages>(&db, group_size, &mut writer)?;
-    write::<ContractsRawCode>(&db, group_size, &mut writer)?;
-    write::<ContractsLatestUtxo>(&db, group_size, &mut writer)?;
-    write::<ContractsState>(&db, group_size, &mut writer)?;
-    write::<ContractsAssets>(&db, group_size, &mut writer)?;
+    let db = combined_db.on_chain();
+    write::<Coins, OnChain>(db, group_size, &mut writer)?;
+    write::<Messages, OnChain>(db, group_size, &mut writer)?;
+    write::<ContractsRawCode, OnChain>(db, group_size, &mut writer)?;
+    write::<ContractsLatestUtxo, OnChain>(db, group_size, &mut writer)?;
+    write::<ContractsState, OnChain>(db, group_size, &mut writer)?;
+    write::<ContractsAssets, OnChain>(db, group_size, &mut writer)?;
+    write::<Transactions, OnChain>(db, group_size, &mut writer)?;
 
-    let block = db.on_chain().latest_block()?;
+    let db = combined_db.off_chain();
+    write::<TransactionStatuses, OffChain>(db, group_size, &mut writer)?;
+    write::<OwnedTransactions, OffChain>(db, group_size, &mut writer)?;
```

Comment:
Maybe we need to create an issue for the future to parallelize snapshot creation=)

Reply:
https://github.com/FuelLabs/fuel-core/issues/1789